### PR TITLE
Slot session creation page

### PR DESCRIPTION
**Pull request by kariy**
Screenshot 2024-06-18 at 3 09 57 PM

on success:

Screenshot 2024-06-18 at 3 10 39 PM

on failure:
Screenshot 2024-06-18 at 3 12 10 PM

### CODE REVIEW

**Comment by broody** [comment]
Getting a few hydration errors on slot/session page

image

File: packages/keychain/src/pages/slot/session/index.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+      fetch(url, {
+        body,
+        headers,
+        method: "POST",
+      })
+        .then((res) => {
+          res.status === 200
+            ? router.replace(`/slot/auth/success`)
+            : new Promise((_, reject) => reject(res));
```
Comment (broody) [inline packages/keychain/src/pages/slot/session/index.tsx:50]: I think we need to `throw new Error()` here, otherwise this is uncaught

**Review by broody — COMMENTED** [review]
Minor comment, otherwise works great!

File: packages/keychain/src/pages/slot/session/index.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      const url = decodeURIComponent(queries.callback_uri);
+      const body = JSON.stringify(session);
+      const headers = new Headers();
+      headers.append("Content-Type", "application/json");
+
+      fetch(url, {
+        body,
+        headers,
+        method: "POST",
+      })
```
[the remark this answers is not in the document]
Comment (broody) [reply packages/keychain/src/pages/slot/session/index.tsx:46]: Would be good to fix this too

File: packages/keychain/src/pages/slot/session/index.tsx (hunk 1 above)
Comment (kariy) [reply packages/keychain/src/pages/slot/session/index.tsx:50]: i believe it should be caught with chain'd '.catch()`, no?

Comment (broody) [reply packages/keychain/src/pages/slot/session/index.tsx:50]: Mmm if u try to make it fail, there's an uncaught error message. You could also do `return Promise.reject()`

Comment (kariy) [reply packages/keychain/src/pages/slot/session/index.tsx:50]: ok i just have to make it a return statement, fixed it https://github.com/cartridge-gg/controller/pull/369/commits/bb3b07eee519d713d3260dc59baf6d29dc8a68e9 .

**Comment by kariy** [comment]
> Getting a few hydration errors on slot/session page
> 
> image

fixed in 4a7f7f82cff017cd4074ccb0a7c5cc7d1a57ce1e

File: packages/keychain/src/pages/slot/session/index.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
```
[the remark this answers is not in the document]
Comment (kariy) [reply packages/keychain/src/pages/slot/session/index.tsx:51]: @broody i think the callback url here will always be susceptible to user forgery (unless the full url is hardcoded, right it's not bcs ew rely on random port number). so i added a bouncer to validate the callback url here [6e98e30](https://github.com/cartridge-gg/controller/pull/369/commits/6e98e304e96287a0f6463e9ff0daf448ed8b0bda). Let me know if there's better alternative than this.

[the remark this answers is not in the document]
Comment (broody) [reply packages/keychain/src/pages/slot/session/index.tsx:51]: Ah make sense!

**Pull request merged**